src/core/ddsc/src/dds_rhc_default.c

```diff
@@ -855,6 +917,14 @@ static bool add_sample (struct dds_rhc_default *rhc, struct rhc_instance *inst,
      counts - are included in the sample).  While this would be place to do it,
      we do it later to avoid having to roll back on allocation failure */
 
+#ifdef DDSI_INCLUDE_DEADLINE_MISSED
+  /* receiving a sample from a writer always results in renewing the
```

Comment:
I wonder whether considering the arrival of a sample that doesn't get stored as sufficient reason to do this makes sense. The reader-side deadline is for monitoring whether the reader's instance got updated; if it doesn't get stored (for whatever reason), arguably it hasn't been updated. To me it makes more sense to do it if and only if the instance really has been updated.